avionics/tests/unit_States.cpp

```diff
@@ -10,6 +10,17 @@ struct Ignitor {
 #include "states/ascent_to_apogee.h"
 #include "states/repeated_checks.hpp"
 
+// Mocking time.cpp
+namespace Hal {
+    t_point now_ms() { 
+        std::chrono::steady_clock::time_point now = std::chrono::steady_clock::now();
+        return std::chrono::time_point_cast<std::chrono::milliseconds>(now); 
+    }
+    uint32_t millis() { return static_cast<uint32_t>(now_ms().time_since_epoch().count()); }
+
+    void sleep_ms(uint32_t t) {std::this_thread::sleep_for(std::chrono::milliseconds(t));}
```

Comment:
Inefficient to require the unit tests to actually sleep; unit tests should be fast. You should probably mock the clock to instead have a current time variable, and sleep() just increments the time (and immediately returns). For that matter the current time can be a public member variable, just to simplify things